### DISCUSSION TITLE
Support partition key in Milvus

### DISCUSSION THREAD

**Discussion by ziyi-curiousthing** [zilliztech/GPTCache#602] [Ideas]
Hi,

We are building a cache layer for openAI backend in our solution and currently experimenting with GPTCache. We used mysql + milvus (zilliz) as our datastore. In our solution, we have the requirement to refine the search space of the cache, e.g. search only part of the cache matching some criteria. We find one way to do it -- initialise a number of cache instances using different datastores to divide the cache, e.g. different tables in mysql + a separate collection in zilliz. However, we hit number of collections limitation in zilliz (256 collections per cluster). We have an idea to utilise the partition feature of milvus to reduce the number of collections needed, so that we can search with the partition key. However, this will require an update in gptcache. Do you think it is feasible and acceptable to do it? If yes, we can work on the feature and potentially send you a pull request.

**Comment by JackLCL** [comment]
@ziyi-curiousthing 
Hi ziyi, about updating gptcache to support milvus partition key, it seems to be a meaningful requirement. Can we arrange an online meeting for more in-depth communication? My email is chenglong.li@zilliz.com, looking forward to your reply.

**Reply by ziyi-curiousthing** [reply]
Hi @JackLCL , thank you for your prompt reply. Would love to have a discussion with you. I sent an email to you. In the meanwhile, I built a proof of concept, the change seems to be quite straightforward.
